libvast/include/vast/concept/convertible/data.hpp

```diff
@@ -89,22 +91,29 @@ caf::error insert_to_map(To& dst, typename To::key_type&& key,
 // clang-format on
 
 template <class... Args>
+// TODO in future CAF. Maybe it will be allowed to change the context of
+// caf::error instead of creating a new object
 [[nodiscard]] caf::error
 prepend(caf::error&& in, const char* fstring, Args&&... args) {
-  if (in) {
-    auto new_msg = in.context().apply({
-      [&, f = fmt::format("{}{{}}", fstring)](std::string& s) {
-        return caf::make_message(fmt::format(
-          VAST_FMT_RUNTIME(f), std::forward<Args>(args)..., std::move(s)));
-      },
-    });
-    if (new_msg)
-      in.context() = std::move(*new_msg);
-    else
-      in.context() = caf::make_message(
-        fmt::format(VAST_FMT_RUNTIME(fstring), std::forward<Args>(args)...));
+  if (!in)
+    return std::move(in);
+  if (in.category() != caf::type_id_v<vast::ec>) {
+    VAST_WARN("Prepend got non vast error: {}. Returning original error", in);
```

Comment:
I don't think this is a user-facing error. Can we make this an assertion for this function instead and then just assume below?